CMakeLists.txt

```diff
@@ -49,7 +49,7 @@ add_library(urcl SHARED
 )
 add_library(ur_client_library::urcl ALIAS urcl)
 target_compile_options(urcl PRIVATE -Wall -Wextra -Wno-unused-parameter)
-target_compile_options(urcl PUBLIC ${CXX17_FLAG})
+target_compile_options(urcl PRIVATE ${CXX17_FLAG})
```

Comment:
What was your motivation for that switch? In my point of view it does make sense to export this to downstream packages.

A better approach would be to expose [`target_compile_features`](https://cmake.org/cmake/help/latest/manual/cmake-compile-features.7.html#requiring-language-standards) but I don't really see this relevant in the scope of this PR.